app/presenters/grouped_result_set_presenter.rb

```diff
@@ -27,45 +24,69 @@ def grouped_documents
     # by the primary facet, then add the document to default group to prevent
     # duplication in every primary facet value grouping.
     default_documents, other_documents = sorted_documents.partition do |document|
-      filtered_by_primary_facet? && tagged_to_all?(primary_facet_key, document)
+      selected_values_in_primary_facet.present? && tagged_to_all?(primary_facet_key, document)
     end
 
     default_group = default_documents.empty? ? [] : [{ facet_key: "all_businesses", facet_name: "All businesses", documents: default_documents }]
 
-    other_documents.each do |item|
-      document_metadata = item[:document][:metadata]
-      document_metadata.each do |metadata|
-        key = metadata[:id]
-        next unless key && facet_filters.has_key?(key.to_sym)
-
-        if primary_facet_key == key
-          # Group by value for the primary facet
-          (metadata[:labels] & facet_filters[primary_facet_key.to_sym]).each do |value|
-            populate_group(primary_group, value, facet_label_for(value), item)
-          end
-        else
-          # Group by facet name otherwise
-          populate_group(secondary_group, key, metadata[:label], item)
-        end
-      end
+    unsorted_primary_group = selected_values_in_primary_facet.map do |selected_value|
+      documents = documents_tagged_to_primary_facet_value(other_documents, selected_value)
+      {
+        facet_key: selected_value,
+        facet_name: label_for_facet_value(selected_value),
+        documents: documents
+      }
+    end
+
+    primary_group = unsorted_primary_group.sort_by { |group| group[:group_name] }
```

Comment:
Should this be sorting by `:facet_name`?

Reply:
I renamed facet_name to group_name, which I think makes more sense because they are either values of the primary facet or the name of a secondary facet - but what they have in common is that they are the name of a grouping